pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -224,6 +224,8 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 				return nil, fmt.Errorf("running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false. /proc/swaps contained: %v", swapLines)
 			}
 		}
+	} else if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) && !cgroups.IsCgroup2UnifiedMode() {
```

Comment:
ok , there is some complexity here, there is an option that defines the behavior of the kubelet if swap is enabled or no

```
// failSwapOn tells the Kubelet to fail to start if swap is enabled on the node.
--
581 | // Default: true
582 | // +optional
583 | FailSwapOn *bool `json:"failSwapOn,omitempty"`
```

if that option is not enabled, the kubelet keeps working.

Now we are going to add a new failure mode, if the feature gate `NodeSwap` is enabled and cgroupsv1, it has to fail too.

The question here is that featuregates are not flags, so they can not be used to define conditional behaviors, when the feature graduates the feature gate will no longer be available and will make the kubelet to always fail on hosts with cgroupsv1 and swap enabled but the "FailSwapOn" to false, that will be confusing

/cc @SergeyKanzhelev

Reply:
IMO, I think this should be fine as is.

Even though swap is in beta (technically beta2), we are rolling this out with the feature flag off. Its a bit more involved feature as the node needs to have swap set up correctly to work. As with any feature we will eventually drop the feature gate when we GA but I think we need a lot more soak time for this feature.

Reply:
those are the right questions Kevin, and that is the goal of the KEPs , you discuss this ambiguous problems and reach a conclusion ... and there is a place to go back and see "oh, this was done because of this", there is a section already for these kind of problems that PRR reviewers check , cc: @wojtek-t

Reply:
I see @SergeyKanzhelev also did similar comments in the previous PR https://github.com/kubernetes/kubernetes/pull/119327#discussion_r1265848566

Reply:
I decided to not use the feature toggle and I use the config option `--fail-swap-on=false` as that will be required to run kubelet on swap.

We are still debating about how swap rollout should go so I think this is fine now.

/cc @harche